go/integration/pkcs11kms/README.md

```diff
@@ -0,0 +1,99 @@
+# PKCS11 HSM/Token support
+
+This integration allows for the AEAD to be used by Tink to be stored in a PKCS11 device.
+
+## Setup
+
+### Provide a *crypto11.Config
+
+As this is focused on support HSMs, there is an assumption that you have at least configured your needed PKCS11 library/client, and generared a Slot/PIN for authentication to the device.
+
+You must provide a *crypto11.Config pointer that has the Library, PIN, and Slot or Label to authenticate with.  
+
+For more details on configuration and usage of the crypto11 package, please refer to the project here: [https://github.com/ThalesIgnite/crypto11](https://github.com/ThalesIgnite/crypto11)
+
+
+## Usage - HSM or Wrapped Key
+
+This KMSClient supports HSM AEAD and Wrapped AEAD keys for the GetAEAD() function.  
+
+Both are using AES256GCM to perform JWK/JWE based encryption operations using GOSE [https://github.com/ThalesIgnite/gose](https://github.com/ThalesIgnite/gose) which are optimized helpers for all things JOSE.  
+
+### HSM
+
+HSM Keys reside in the pkcs11 token/slot, and a cipher.AEAD handle to the PKCS11 key handle which is then used to support the tink.AEAD interface transparently.
+
+HSM Keys are:
+- SecretKeys in the HSM
+- Session based using a pkcs11 session to the HSM for each KMSClient provided
+- The AEAD handle will be accessible as long as the session is open
+- Good for "remove the HSM" use cases that will close all access to the AEAD even to tink
+
+The keyURI for a HSM key is `pkcs11://ee2c5de3-335c-428b-8168-5daf299e97f3`  where `ee2c5de3-335c-428b-8168-5daf299e97f3` is the key ID (not label) in the device to use.  To have the key auto created if it doesnt' exist yet, simply proved the "autogen" bool to the client.
```

Comment:
change:
simply `proved` the "autogen" bool to the client. 
==>
simply `provide` the "autogen" bool to the client.

Reply:
good catch